plugins/woocommerce/includes/export/abstract-wc-csv-batch-exporter.php

```diff
@@ -130,7 +130,7 @@ protected function write_csv_data( $data ) {
 			return false;
 		}
 
-		$fp = fopen( $this->get_file_path(), 'a+' );
+		$fp = fopen( $this->get_file_path(), apply_filters( 'woocommerce_fopen_csv', 'a+' ) );
```

Comment:
I wonder if this name is a little too terse. Something like `woocommerce_csv_exporter_fopen_mode` may be better:

- `csv_exporter` is a little more specific than just `csv` (noting that we also perform a `fopen()` call in the CSV **importer** code, which this will not impact)
- `mode` speaks to the specific thing we are allowing adjustments for

What are your thoughts? Note, we would also ideally [have a docblock](https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#4-hooks-actions-and-filters) for any new hooks like this one.

Reply:
Ye think the name change makes more sense, I like the suggestion, will add docblock and update the PR